presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -95,8 +97,8 @@ public SubPlan createSubPlans(Session session, Plan plan, boolean forceSingleNod
         PlanNode root = SimplePlanRewriter.rewriteWith(fragmenter, plan.getRoot(), properties);
 
         SubPlan subPlan = fragmenter.buildRootFragment(root, properties);
-        subPlan = analyzeGroupedExecution(session, subPlan);
         subPlan = reassignPartitioningHandleIfNecessary(session, subPlan);
+        subPlan = analyzeGroupedExecution(session, subPlan);
```

Comment:
Curious: Why do we change the execution sequence here?

Reply:
Now in `analyzeGroupedExecution`, it requires the partitioning of the fragment to get the `BucketNodeMap`. Thus we need to decide the fragment partitioning prior to analyzing grouped execution.